app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt

```diff
@@ -78,6 +96,61 @@ class MapViewModel : ViewModel() {
     _selectedLocation.value = location
   }
 
+  /**
+   * h Get the bottom left and top right corners of the screen in latitude and longitude
+   * coordinates. The corners are calculated based on the center of the screen and the viewport
+   * dimensions. If useBuffer is true, the corners are calculated with a buffer of 2x the viewport
+   * dimensions. This is useful for loading parkings that are not yet visible on the screen.
+   *
+   * @param mapView the MapView to get the screen corners from
+   * @return a pair of the bottom left and top right corners of the screen
+   */
+  fun getScreenCorners(mapView: MapView): Pair<Point, Point> {
+    // Retrieve viewport dimensions
+    val viewportWidth = mapView.width
+    val viewportHeight = mapView.height
+
+    val centerPixel = mapView.mapboxMap.pixelForCoordinate(mapView.mapboxMap.cameraState.center)
+
+    // Calculate the multiplier for the buffer
+    val multiplier = 3.0
+
+    val bottomLeftCorner =
+        mapView.mapboxMap.coordinateForPixel(
+            ScreenCoordinate(
+                centerPixel.x - (viewportWidth * multiplier),
+                centerPixel.y + (viewportHeight * multiplier)))
+
+    val topRightCorner =
+        mapView.mapboxMap.coordinateForPixel(
+            ScreenCoordinate(
+                centerPixel.x + (viewportWidth * multiplier),
+                centerPixel.y - (viewportHeight * multiplier)))
+
+    return Pair(bottomLeftCorner, topRightCorner)
+  }
+
+  /**
+   * Initialize the location component of the map.
+   *
+   * @param mapView the MapView to initialize the location component on
+   * @param mapViewModel the MapViewModel to update the camera position
+   */
+  fun initLocationComponent(mapView: MapView, mapViewModel: MapViewModel) {
+    val locationComponentPlugin = mapView.location
+    locationComponentPlugin.updateSettings {
+      this.enabled = true
+      this.locationPuck = createDefault2DPuck(true)
+    }
+    val onIndicatorPositionChangedListener = OnIndicatorPositionChangedListener { location ->
+      mapView.mapboxMap.setCamera(CameraOptions.Builder().center(location).build())
+      mapViewModel.updateCameraPosition(mapView.mapboxMap.cameraState)
+    }
+
+    locationComponentPlugin.addOnIndicatorPositionChangedListener(
+        onIndicatorPositionChangedListener)
```

Comment:
[Non-blocking note] In general, I would avoid having a such intermediary variable, as they tend to make the code harder to read. In my opinion, it would have been fine to go directly for
```
locationComponentPlugin.addOnIndicatorPositionChangedListener( { location ->
        mapView.mapboxMap.setCamera(CameraOptions.Builder().center(location.build())
        mapViewModel.updateCameraPosition(mapView.mapboxMap.camera)
    }
)

Reply:
fixed thanks